src/test/java/fi/aalto/cs/apluscourses/intellij/model/SettingsImporterImplTest.java

```diff
@@ -0,0 +1,34 @@
+package fi.aalto.cs.apluscourses.intellij.model;
+
+import com.intellij.openapi.util.io.FileUtilRt;
+import fi.aalto.cs.apluscourses.utils.DomUtil;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+
+public class SettingsImporterImplTest {
+
+  @Test
+  public void testCreateCustomWorkspaceXml() throws IOException {
```

Comment:
:sunglasses: if only there could be more cases covered, as the original method seems to have quite a branching logics?

Reply:
Added a test case for when `worspace.xml` already contains the setting